muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/discussion/body/application/current-thread.tsx

```diff
@@ -98,13 +98,24 @@ class CurrentThread extends React.Component<CurrentThreadProps, CurrentThreadSta
       avatar = <div className="avatar avatar--category-1"></div>;
     } else {
       //This is what it shows when the user is ready
-      avatar = <Avatar key={userCreator.id} id={userCreator.id} firstName={userCreator.firstName} hasImage={userCreator.hasImage} userCategory={userCategory} avatarAriaLabel={this.props.i18n.text.get("plugin.wcag.userAvatar.label")}/>
+      avatar = <Avatar key={userCreator.id} id={userCreator.id} firstName={userCreator.firstName} hasImage={userCreator.hasImage} userCategory={userCategory} avatarAriaLabel={this.props.i18n.text.get("plugin.wcag.userAvatar.label")} />
     }
 
-    const canRemoveThread = this.props.userId === this.props.discussion.current.creator.id || areaPermissions.removeThread;
-    const canEditThread = this.props.userId === this.props.discussion.current.creator.id || areaPermissions.editMessages;
-    const threadLocked = this.props.discussion.current.locked === true;
-    const student = this.props.status.isStudent === true;
+    const student: boolean = this.props.status.isStudent === true;
+    const threadOwner: boolean = this.props.userId === this.props.discussion.current.creator.id
+    const canRemoveThread: boolean = !student && threadOwner || areaPermissions.removeThread;
```

Comment:
This can read ambiguously, I think we'd rather add parenthesis to them; technically the and operator has precedence; so you can figure it out, but it's hard to read if you forget.

Bit of a nitpick but I say let's add some parenthesis for readability.